extends CanvasLayer

class_name Message

# refer√™cia ao label de texto
@onready var label = get_node("Label") as Label;
# refer√™ncia ao timer de mensagem
@onready var timerOfTheMessage = get_node("Timer") as Timer;

# dicion√°rio de mensagens de erros e acertos
var messages : Dictionary = {
	"correctUnion" : [
		"üíò Amor √† primeira vista!",
		"üíò Voc√™ uniu um casal perfeito!",
		"üíò Cupido arrasou!",
		"üíò Louca paix√£o!",
		"üíò Cupido n√£o erra!",
		"üíò Amor verdadeiro detectado!",
		"üíò Que casal lindo!",
		"üíò Felicidade garantida!",
		"üíò Mais um casal feliz!",
		"üíò Amor eterno garantido!"
	],
	"wrongUnion" : [
		"‚ùå Ops, tente novamente!",
		"‚ùå Esse n√£o √© o par certo!",
		"‚ùå Cuidado, voc√™ errou!",
		"‚ùå Possue um outro amor",
		"‚ùå N√£o foi dessa vez!",
		"‚ùå Continue tentando!",
		"‚ùå Esse par n√£o deu certo!",
		"‚ùå Uni√£o falhou!",
		"‚ùå Voc√™ errou, mas n√£o desista!",
		"‚ùå Esse par n√£o combina!"
	]
}

func _ready() -> void:
	#Global.shootNode.unionCorrect.connect(emitMessage)
	Global.message = self;


## fun√ß√£o para exibir uma mensagem aleat√≥ria de um vetorde mensagens
func emitMessage(list : Array) -> void:
	randomize();
	# pega valor aleat√≥rio no range do letor
	var _index = randi_range(0, list.size() -1);
	# pega a mensagem referente a posi√ß√£o do index aleat√≥rio sorteado
	var _message = list[_index];
	# joga o conte√∫do da mensagem no campo de texto do label
	label.text = _message;
	# ativa a visibilidade do label
	label.visible = true;
	timerOfTheMessage.start();
	

## fun√ß√£o executada quando o tempo da mensagem chegar ao fim
func _on_timer_timeout():
	# desabilita a visibilidade do label
	label.visible = false;
